esrally/mechanic/launcher.py

```diff
@@ -175,6 +175,9 @@ def _prepare_env(self, car_env, node_name, java_home, t):
             self._set_env(env, "PATH", os.path.join(java_home, "bin"), separator=os.pathsep, prepend=True)
             # Don't merge here!
             env["JAVA_HOME"] = java_home
+        elif env.get("JAVA_HOME"):
```

Comment:
Should we take a cleaner (and bolder) approach and not inherit anything from the env apart from PATH?

Practically speaking I thought about replacing lines 172 and 172 with something like:
`env={k:v for k, v in os.environ.items() if k == "PATH"}`

TBH I can't think of what we need from the shell PATH to launch ES, we might just as well be ok with removing line 172 altogether, but typically all launched Elasticsearches (even in Docker) will have something in PATH so this sounds ok to me.

Reply:
I like that approach as it is much cleaner. I can think of situations when we want to set other environment variables like `LD_LIBRARY_PATH` in order set a custom allocator like [jemalloc](https://github.com/jemalloc/jemalloc) to use it for [analyzing native memory leaks](https://github.com/jemalloc/jemalloc/wiki/Use-Case%3A-Leak-Checking). 

I suggest that we leverage our config object here to only pass `PATH` by default but allow to override it with a config property. We should avoid using the config object everywhere and instead derive a property in the constructor of `ProcessLauncher` as follows:

```python
from esrally.utils import opts
#...
self.pass_env_vars = opts.csv_to_list(self.cfg.opts("system", "passenv", mandatory=False, default_value="PATH"))
```

Reply:
Do we want to overwriting `ES_JAVA_OPTS`? Or append the ExitOnOOM to it? Ill leave it currently as overwriting, unless we want to change the behaviour here.

Reply:
After chatting w @dliappis, we will honor the user's `ES_JAVA_OPTS` if provided in the rally.ini, otherwise we will add the flag.

Reply:
Do we want to always glean the `PATH` even if it is not specified in the config? As is if the user adds variables, they might forget the path and then :boom:

Reply:
I tend not to apply any special logic for `PATH` and would favor if the user would clearly see what environment variables are passed. I also expect that the default will only be overridden in very special circumstances (like the example I've provided above).

Reply:
ok I will remove the PATH special case!